client/src/main/kotlin/io/spine/examples/shareaware/client/wallet/WalletPage.kt

```diff
@@ -105,15 +108,109 @@ fun WalletPage() = Column {
+        var replenishmentAmount by remember { mutableStateOf("") }
+        if (replenishmentState) {
+            MoneyOperationDialog(
+                onCancel = { replenishmentState = false },
+                onConfirm = {},
+                title = "Wallet Replenishment",
+                ibanValue = replenishmentIbanValue,
+                onIbanChange = { replenishmentIbanValue = it },
+                moneyValue = replenishmentAmount,
+                onMoneyChange = { replenishmentAmount = it }
+            )
+        }
+        var withdrawalIbanValue by remember { mutableStateOf("") }
+        var withdrawalAmount by remember { mutableStateOf("") }
+        if (withdrawalState) {
+            MoneyOperationDialog(
+                onCancel = { withdrawalState = false },
+                onConfirm = {},
+                title = "Wallet Withdrawal",
+                ibanValue = withdrawalIbanValue,
+                onIbanChange = { withdrawalIbanValue = it },
+                moneyValue = withdrawalAmount,
+                onMoneyChange = { withdrawalAmount = it }
+            )
+        }
     }
 }
+
+/**
+ * Dialog window component with a from for money operations.
+ *
+ * @param ibanValue the IBAN value to be shown in the relevant input
+ * @param onIbanChange the callback that is triggered when the IBAN value change
+ * @param moneyValue the money amount value to be shown in the relevant input
+ * @param onMoneyChange the callback that is triggered when the money amount value change
+ *
+ * @see Dialog
+ */
+@Composable
+private fun MoneyOperationDialog(
+    onCancel: () -> Unit,
+    onConfirm: () -> Unit,
+    title: String,
+    ibanValue: String,
+    onIbanChange: (String) -> Unit,
+    moneyValue: String,
+    onMoneyChange: (String) -> Unit
+) {
+    var mistakeInIbanField by remember { mutableStateOf(false) }
+    var mistakeInMoneyField by remember { mutableStateOf(false) }
+    Dialog(
+        onCancel = onCancel,
+        onConfirm = {
+            mistakeInIbanField = ibanValue.validateIban()
+            mistakeInMoneyField = moneyValue.validateMoney()
+            if (!mistakeInIbanField && !mistakeInMoneyField) {
+                onConfirm()
+                onCancel()
+            }
+        },
+        title = title,
+        {
+            Input(
+                value = ibanValue,
+                onValueChange = onIbanChange,
+                label = "Please enter your IBAN",
+                isError = mistakeInIbanField,
+                errorMessage = "Please ensure that your IBAN " +
+                        "contains 2 letters and 2 digits in the beginning and " +
+                        "up to 26 alphanumeric characters after." +
+                        "Example: FI211234569876543210"
+            )
+        },
+        {
+            Input(
+                value = moneyValue,
+                onValueChange = onMoneyChange,
+                label = "Please enter money amount",
```

Comment:
No "please" in messages. "IBAN must be properly formatted."

As for placeholders guiding the user:

* "Please enter your IBAN" should be just "IBAN".
* "Please enter money amount" should be "How much".
And the latter field should probably indicate the currency.